### PR TITLE
feat: #6 create admin request management page

### PR DESCRIPTION
**Pull request by nevence**
created HR (admin) and Employee request management pages, implemented filtering by leave type, status, search by name, and page size 
implemented UI and logic for requests approval and rejection, and adding (editing) comments (HR and Employee) / changing leave type (HR)
Screenshot 2024-06-19 020929
Screenshot 2024-06-19 020942
Screenshot 2024-06-19 020957
Screenshot 2024-06-19 021044
Screenshot 2024-06-19 021114
Screenshot 2024-06-16 150247
Screenshot 2024-06-16 150304
Screenshot 2024-06-16 150326
Screenshot 2024-06-16 150407
Screenshot 2024-06-16 150448
Screenshot 2024-06-16 150506
Screenshot 2024-06-16 150532
Screenshot 2024-06-16 150546
Screenshot 2024-06-16 150606
Screenshot 2024-06-16 150828
Screenshot 2024-06-16 152442

### CODE REVIEW

**Comment by nevence** [comment]
[Admin request management page](https://trello.com/c/9zSr6HxD/6-admin-request-management-page)

**Review by pdavitkovic — APPROVED** [review]

File: VacaYAY/BusinessLogicLayer/Extensions/VacationRequestExtension.cs [hunk 1]
```diff
@@ -56,6 +57,19 @@ public static void MapToVacationRequestUpdate(this VacationRequest vacationReque
...
+        public static void MapVacationRequestReject(this VacationRequest vacationRequest)
+        {
+            vacationRequest.Status = Enums.VacationRequestStatus.Rejected;
+            vacationRequest.UpdateDate = DateTime.UtcNow;
+        }
+
+        public static void MapVacationRequestApprove(this VacationRequest vacationRequest)
+        {
+            vacationRequest.Status = Enums.VacationRequestStatus.Accepted;
+            vacationRequest.Employee.DaysOffNumber -= (vacationRequest.EndDate - vacationRequest.StartDate).Days;
```
Comment (NinaLazarevicIng) [inline VacaYAY/BusinessLogicLayer/Extensions/VacationRequestExtension.cs]: Days off for employee are also number of working days, so you can use your function GetWorkingDays here.

Comment (nevence) [reply VacaYAY/BusinessLogicLayer/Extensions/VacationRequestExtension.cs]: great observation, i accidentally let that slide

**Comment by nevence** [comment]
[Add pagination](https://trello.com/c/K6T1jF9i/7-add-pagination)

File: VacaYAY/PresentationLayer/ActionFilters/NotificationFilter.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+namespace PresentationLayer.ActionFilters
+{
+    public class NotificationFilter : IResultFilter
+    {
+        public void OnResultExecuted(ResultExecutedContext context)
+        {
+        }
+
+        public void OnResultExecuting(ResultExecutingContext context)
+        {
```
Comment (pdavitkovic) [inline VacaYAY/PresentationLayer/ActionFilters/NotificationFilter.cs]: Seems like an overkill. 

You can try this: https://www.c-sharpcorner.com/blogs/alert-message-from-controller-view-using-javascript-alert-messagebox

**Review by pdavitkovic — CHANGES_REQUESTED** [review]

**Review by pdavitkovic — APPROVED** [review]

**Pull request merged**